### ISSUE TITLE
-o-n vs -on- (e.g., dekonaĵon)

### ISSUE THREAD

**Issue by ProfessorO** [ProfessorO/esperanta-analizisto#21]
The former is a noun + accusative (direct object) ending, while the latter means "fraction" (e.g., dek = ten, dekono = tenTH). So "dekonaĵon" (Genesis 14:20, tithe) is a challenge. The first -on- is a single root, while the one at the end is actually two roots, -o + -n.

In theory, we could add entries that have -on after all the numbers (duon, trion, kvaron, k.t.p.). However, those aren't roots—they're two parts. Looks like there's some interesting stuff about where roots can appear, as I think -on- can only appear in the middle of a word, while -o and -n can appear at the end.

Do we want to mark certain roots with some restrictions as to where they can appear? Only after numbers, only in the middle of a word, etc.

**Comment by ProfessorO** [comment]
For now I'm leaving "on" out as a root, as it's FAR less common than the -o and -n endings.

**Comment by stephenwade** [comment]
-on- can only appear in the middle of a word.

Can -o and -n also appear in the middle of a word? If so, then we still have no way of knowing whether "on" should be parsed as -on- or -o-n-.

**Comment by ProfessorO** [comment]
-o can definitely appear in the middle of compound words. For example, akvofonto, akv-o-font-o.

-n really should only appear at the very end of a word, as it's the direct-object marker for adjectives and nouns.

**Comment by stephenwade** [comment]
Hm, okay. Could -on appear at the end of a word? If not, we could make rules saying "n" should always be parsed at the end of a word and never parsed otherwise.

**Comment by ProfessorO** [comment]
So the root should be n$ instead of just n (using $ in the PCRE sense). Hmmm. That could work. Could also address the issue with iĝis parsing as i-ĝis instead of iĝ-is (#23), if the root were ^ĝis instead of just ĝis.

**Comment by stephenwade** [comment]
Yeah, I like that idea. I think we might also find priorities useful, and parse longest-first in each priority (set "n$" to be a higher priority than others, so that it's always parsed when it matches instead of being overridden by longer roots).
